mage_ai/frontend/components/PipelineDetail/AddNewBlocks/index.tsx

```diff
@@ -38,23 +39,41 @@ function AddNewBlocks({
   const [buttonMenuOpenIndex, setButtonMenuOpenIndex] = useState(null);
   const dataLoaderButtonRef = useRef(null);
   const transformerButtonRef = useRef(null);
+  const dataExporterButtonRef = useRef(null);
   const sharedProps = {
     compact,
     inline: true,
   };
 
-  const dataLoaderMenuItems = DATA_SOURCE_TYPES.map((sourceType: DataSourceTypeEnum) => ({
-    label: () => DATA_SOURCE_TYPE_HUMAN_READABLE_NAME_MAPPING[sourceType],
-    onClick: () => {
-      addNewBlock({
-        config: {
-          data_source: sourceType === DataSourceTypeEnum.GENERIC ? null : sourceType,
-        },
-        type: BlockTypeEnum.DATA_LOADER,
-      });
-    },
-    uuid: sourceType,
-  }));
+  const dataLoaderMenuItems = (
+    DATA_SOURCE_TYPES[BlockTypeEnum.DATA_LOADER].map((sourceType: DataSourceTypeEnum) => ({
+      label: () => DATA_SOURCE_TYPE_HUMAN_READABLE_NAME_MAPPING[sourceType],
+      onClick: () => {
+        addNewBlock({
+          config: {
+            data_source: sourceType === DataSourceTypeEnum.GENERIC ? null : sourceType,
+          },
+          type: BlockTypeEnum.DATA_LOADER,
+        });
+      },
+      uuid: `data_loader/${sourceType}`,
+    }))
+  );
+
+  const dataExporterMenuItems = (
```

Comment:
There's a bunch of duplicate code in `dataExporterMenuItems` and `dataLoaderMenuItems`. Try abstracting some of it out or create a util method in the `utils.ts` file to avoid some of the duplicate code.

Reply:
Was thinking about this, I'll also add in the data sources to transformer blocks using that util method and then rerequest review